### ISSUE TITLE
[FR] PUI - Add option for plugins to add header actions

### ISSUE THREAD

**Issue by matmair** [inventree/InvenTree#8593]
### Please verify that this feature request has NOT been suggested before.

- [x] I checked and didn't find a similar feature request

### Problem statement

Plugins do not have a way to add quick action to the header action of items. Therefore users need to click at least twice to issue a plugin action once they are on the page for the item.

### Suggested solution

Add a way for plugins to expose 0 or more custom actions for (specific) items
![Image](https://github.com/user-attachments/assets/6209ceaf-3f24-4609-8e46-3e5e23a69708)
They should at least be able to set:
- Text
- Icon
- Tooltip text

To enable some transparency I think all plugin action should have a standard information from which plugin the action is emitted added to its tooltip

### Describe alternatives you've considered

Adding a custom panel

### Do you want to develop this?

- [x] I want to develop this.

**Comment by matmair** [comment]
@wolflu05 any input on this?

**Comment by wolflu05** [comment]
Yes, that's exactly what I was planning to add with the custom UI feature system. I have added some more detail on how I imagined the system in https://github.com/inventree/InvenTree/pull/8278#discussion_r1862652208

**Comment by SchrodingersGat** [comment]
@matmair this would be a great addition :)
